pkg/azureutils/azure_disk_utils.go

```diff
@@ -426,6 +501,101 @@ func NormalizeStorageAccountType(storageAccountType, cloud string, disableAzureS
 	return "", fmt.Errorf("azureDisk - %s is not supported sku/storageaccounttype. Supported values are %s", storageAccountType, supportedSkuNames)
 }
 
+func ParseDiskParameters(parameters map[string]string) (DiskParameters, error) {
```

Comment:
this function could not be combined for CreateVolume, CreateSnapshot and ExpandVolume since different CSI interface accepts different parameters, cannot use one function to validate all parameters.

Reply:
This function was called in CreateVolume and CreateSnapshot functions only which as far as I could tell were validating a subset of the same parameters. I have reverted the change to CreateSnapshot, so this parsing is now only used for CreateVolume.